eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/implementation/ReactorConnectionTest.java

```diff
@@ -248,18 +250,42 @@ public void onConnectionStateOpen() {
      */
     @Test
     public void createCBSNode() {
+        // Arrange
+        final Connection connection = mock(Connection.class);
+        when(connection.getRemoteState()).thenReturn(EndpointState.ACTIVE);
+        final Event mock = mock(Event.class);
+        when(mock.getConnection()).thenReturn(connection);
+        handler.onConnectionRemoteOpen(mock);
+
         // Act and Assert
-        StepVerifier.create(connection.getCBSNode())
+        StepVerifier.create(this.connection.getCBSNode())
             .assertNext(node -> {
                 Assert.assertTrue(node instanceof CBSChannel);
             }).verifyComplete();
     }
 
+    /**
+     * Verifies that if the connection cannot be created within the timeout period, it errors.
+     */
+    @Test
+    public void createCBSNodeTimeoutException() {
+        // Arrange
+        Duration timeout = Duration.ofSeconds(5);
+        ConnectionParameters parameters = new ConnectionParameters(CREDENTIAL_INFO, timeout, tokenProvider,
+            TransportType.AMQP, Retry.getDefaultRetry(), ProxyConfiguration.SYSTEM_DEFAULTS, Schedulers.elastic());
+        ReactorConnection connection = new ReactorConnection(CONNECTION_ID, parameters, reactorProvider, reactorHandlerProvider, mock(AmqpResponseMapper.class));
+
+        // Act and Assert
+        StepVerifier.create(connection.getCBSNode())
+            .verifyError(TimeoutException.class);
+    }
+
+
     /**
      * Verifies if the ConnectionHandler transport fails, then we are unable to create the CBS node or sessions.
      */
     @Test
-    public void cannotCreateResourcesWhenErrored() {
+    public void cannotCreateResourcesWhenErrors() {
```

Comment:
This reads a bit awkward to me.   Maybe consider something like `cannotCreateResourcesOnFailure` or similar?

Reply:
Fixed.